addons/binding/org.openhab.binding.onewiregpio/src/main/java/org/openhab/binding/onewiregpio/handler/OneWireGPIOHandler.java

```diff
@@ -0,0 +1,155 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onewiregpio.OneWireGPIOBindingConstants;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OneWireGPIOHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Anatol Ogorek - Initial contribution
+ */
+public class OneWireGPIOHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OneWireGPIOHandler.class);
+
+    private String gpioBusFile;
+    private Integer refreshTime;
+
+    public OneWireGPIOHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(TEMPERATURE)) {
+            if (command instanceof RefreshType) {
+                publishSensorValue(channelUID);
+            } else {
+                logger.debug("Command {} is not supported for channel: {}. Supported command: REFRESH", command,
+                        channelUID.getId());
+            }
+        }
+    }
+
+    @Override
+    public void initialize() {
+        checkConfiguration();
+        updateStatus(ThingStatus.UNKNOWN);
+        startAutomaticRefresh();
+    }
+
+    /**
+     * This method checks if the provided configuration is valid.
+     * When invalid parameter is found, default value is assigned.
+     */
+    private void checkConfiguration() {
+        Configuration configuration = getConfig();
+        gpioBusFile = (String) configuration.get(GPIO_BUS_FILE);
+        if (StringUtils.isEmpty(gpioBusFile)) {
+            logger.warn("GPIO_BUS_FILE not set. Please check configuration, and set proper path to w1_slave file.");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "The path to the w1_slave sensor data file is missing.");
+            return;
+        }
+
+        refreshTime = ((Number) configuration.get(REFRESH_TIME)).intValue();
+        if (refreshTime.intValue() <= 0) {
+            logger.warn("Refresh time [{}] is not valid. Falling back to default value: {}.", refreshTime,
+                    DEFAULT_REFRESH_TIME);
+            refreshTime = DEFAULT_REFRESH_TIME;
+        }
+    }
+
+    private void startAutomaticRefresh() {
+        Runnable refresher = new Runnable() {
+            @Override
+            public void run() {
+                List<Channel> channels = getThing().getChannels();
```

Comment:
Isn't there always only a single channel? I hence guess that you can simplify this logic here a lot.

Reply:
Yes, currently there is only one channel, but there are oneWire humidity sensors for example, which might be supported as well, so it is very likely, there will be more channels.
I simply don't have sensor to test it so I didin't implement support, but I think it will be easier to extend it in the future.

Reply:
ok, fine with me